nvtabular/ops.py

```diff
@@ -831,7 +831,7 @@ class JoinExternal(TransformOperator):
         if the data is originally stored on disk.
     """
 
-    default_in = ALL
+    default_in = CAT
```

Comment:
Do we only want to be able to join on CAT columns?

Reply:
We can join CAT columns either way, since this op cannot really do anything by default (we need to know `on` explicitly). Mking this change only affects out ability to use `add_*_preprocess` - However, I completely agree that this *should* be `ALL`, so I'll make that change.